JustSaying/NullMessageResponseLogger.cs

```diff
@@ -0,0 +1,13 @@
+﻿using System;
+using System.Threading.Tasks;
+using JustSaying.AwsTools.MessageHandling;
+using JustSaying.Models;
+
+namespace JustSaying
+{
+    internal class NullMessageResponseLogger : IMessageResponseLogger
+    {
+        public Action<MessageResponse, Message> ResponseLogger { get; set; }
```

Comment:
Also, it might be simpler to just expose these methods as normal method signatures instead of properties of `Action` and `Func`. You could provide a base class that defines no-op implementations of both, and then implementors can just override one or both as they wish.

Reply:
>You should make these actually return non-null instances, so that the calling code doesn't need to do null-conditional evaluations.

They were originally no-op implementations but we need to [detect if they're not set in order to call the synchronous version of the async version isn't provided](https://github.com/justeat/JustSaying/pull/350/files#diff-f6c5b29e7f5c2decc9b2ec6ff6a449c3R92).

Reply:
@JosephWoodward How about only exposing the asynchronous version and then if people want to implement it synchronously they can just return `Task.CompletedTask`?